File: kora-java-kafka/src/main/java/ru/tinkoff/kora/example/kafka/listener/AutoCommitRecordJsonListener.java
package ru.tinkoff.kora.example.kafka.listener;

import org.apache.kafka.clients.consumer.ConsumerRecord;
import ru.tinkoff.kora.common.Component;
import ru.tinkoff.kora.json.common.annotation.Json;
import ru.tinkoff.kora.kafka.common.annotation.KafkaListener;

@Component
public final class AutoCommitRecordJsonListener extends AbstractListener<AutoCommitRecordJsonListener.MyEvent> {

    @Json
    public record MyEvent(String username, int code) {}

    @KafkaListener("kafka.consumer.my-listener")
    void process(ConsumerRecord<String, @Json MyEvent> record) {
        success(record.value());
    }
}

File: examples/kotlin/kora-kotlin-kafka/src/main/kotlin/ru/tinkoff/kora/kotlin/example/kafka/listener/AutoCommitRecordJsonListener.kt
package ru.tinkoff.kora.kotlin.example.kafka.listener

import org.apache.kafka.clients.consumer.Consumer
import org.apache.kafka.clients.consumer.ConsumerRecord
import org.apache.kafka.clients.consumer.ConsumerRecords
import org.apache.kafka.common.header.Headers
import org.apache.kafka.common.serialization.Deserializer
import ru.tinkoff.kora.common.Component
import ru.tinkoff.kora.common.Tag
import ru.tinkoff.kora.json.common.JsonReader
import ru.tinkoff.kora.json.common.annotation.Json
import ru.tinkoff.kora.kafka.common.annotation.KafkaListener
import ru.tinkoff.kora.kafka.common.consumer.telemetry.KafkaConsumerTelemetry

@Component
class AutoCommitRecordJsonListener : AbstractListener<AutoCommitRecordJsonListener.MyEvent>() {
    @Json
    data class MyEvent(val username: String, val code: Int)

    @KafkaListener("kafka.consumer.my-listener")
    fun process(record: ConsumerRecord<String, @Json MyEvent>) {
        success(record.value())
    }
}

